scripts/dts/gen_defines.py

```diff
@@ -435,11 +435,27 @@ def irq_name_alias(irq, cell_name):
             alias += "_" + str2ident(cell_name)
         return alias
 
+    def encode_zephyr_multi_level_irq(irq, irq_num):
+        # See doc/reference/kernel/other/interrupts.rst for details
+        # on how this encoding works
+
+        irq_ctrl = irq.controller
+        # Look for interrupt controller parent until we have none
+        while irq_ctrl.interrupts:
+            irq_num = (irq_num + 1) << 8
+            if "irq" not in irq_ctrl.interrupts[0].specifier:
+                _err("missing 'irq' specifier on {!r}".format(irq_ctrl))
```

Comment:
Could give the path to the binding (`irq_ctrl.binding_path`) instead/as well, since that's where it's missing.

```python
_err("expected 'irq' to appear in '#cells' in {} (for {!r})"
     .format(irq_ctrl.binding_path, irq_ctrl)) 
```

Reply:
Aren't we going to get the yaml from the `{!r}`

Does this feel redundant?:

```
expected 'irq' to appear in '#cells' in /home/galak/git/zephyr/dts/bindings/interrupt-controller/openisa,rv32m1-intmux-ch.yaml (for <Device /soc/intmux@4004f000/interrupt-controller@40 in 'rv32m1_vega_ri5cy.dts.pre.tmp', binding /home/galak/git/zephyr/dts/bindings/interrupt-controller/openisa,rv32m1-intmux-ch.yaml>)
```

Reply:
Forgot that it was already mentioned there.

Maybe it could be

```python
_err("expected binding for {!r} to have 'irq' in '#cells'"
     .format(irq_ctrl))
```

Reply:
changed to that.